File: front/src/lib/api/models/EventDto.ts
import { AccountDto } from "./AccountDto";
import { AddressDto } from "./AddressDto";
import { DemandDto } from "./DemandDto";

export type EventDto = {
  /**
   * @type integer | undefined, int64
   */
  id?: number;
  /**
   * @type string
   */
  description: string;
  organizer?: AccountDto;
  /**
   * @type integer | undefined, int64
   */
  eventTypeId?: number;
  /**
   * @type string
   */
  eventType: string;
  /**
   * @type string
   */
  name: string;
  /**
   * @type string, date-time
   */
  date: Date;
  address: AddressDto;
  /**
   * @type integer, int32
   */
  capacity: number;
  /**
   * @type array | undefined
   */
  participants?: AccountDto[];
  /**
   * @type number, float
   */
  price: number;
  /**
   * @type array | undefined
   */
  demands?: DemandDto[];
  /**
   * @type array
   */
  requirements: string[];
  /**
   * @type boolean
   */
  alcoolAllowed: boolean;
  /**
   * @type object | undefined
   */
  additionalProperties?: {
    [key: string]: {};
  };
};

File: back/src/main/java/com/projet/valatMoslehGrodet/dto/EventDTO.java
package com.projet.valatMoslehGrodet.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EventDTO {
    private Long id;
    @NotNull
    private String description;
    private AccountDTO organizer;
    private Long eventTypeId;
    @NotNull
    private String eventType;
    @NotNull
    private String name;
    @NotNull
    private Date date;
    @NotNull
    private AddressDTO address;
    @NotNull
    private Integer capacity;
    private List<AccountDTO> participants;
    @NotNull
    private Float price;
    private List<DemandDTO> demands;
    @NotNull
    private List<String> requirements;
    @NotNull
    private Boolean alcoolAllowed;
    private Map<String, Object> additionalProperties = new HashMap<>();

    public void setAdditionalProperty(String key, Object value) {
        additionalProperties.put(key, value);
    }

    public Object getAdditionalProperty(String key) {
        return additionalProperties.get(key);
    }
}